src/Components/Web.Extensions/src/InputFile/InputFile.cs

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Components.Rendering;
+using Microsoft.JSInterop;
+
+namespace Microsoft.AspNetCore.Components.Web.Extensions
+{
+    /// <summary>
+    /// A component that wraps the HTML file input element and exposes a <see cref="Stream"/> for each file's contents.
+    /// </summary>
+    public class InputFile : ComponentBase, IInputFileJsCallbacks, IDisposable
+    {
+        private ElementReference _inputFileElement;
+
+        private IJSUnmarshalledRuntime? _jsUnmarshalledRuntime;
+
+        private InputFileJsCallbacksRelay? _jsCallbacksRelay;
+
+        [Inject]
+        private IJSRuntime JSRuntime { get; set; } = default!;
+
+        /// <summary>
+        /// Gets or sets the event callback that will be invoked when the collection of selected files changes.
+        /// </summary>
+        [Parameter]
+        public EventCallback<InputFileChangeEventArgs> OnChange { get; set; }
+
+        /// <summary>
+        /// Gets or sets the maximum chunk size for file data sent over a SignalR circuit.
+        /// This only has an effect when using Blazor Server.
+        /// </summary>
+        [Parameter]
+        public int MaxSignalRChunkSize { get; set; } = 20 * 1024; // SignalR limit is 32K.
```

Comment:
You could add the UnsupportedPlatformAttribute on this property

Reply:
Should these default values be settable from options? There's no point in forcing users to specify this value all the time when they change the underlying global value, since this is not a "per file" setting.